### PR TITLE
MEPTS-1300 | MI Cat 13 Diff In ARIEL DB

### PR DESCRIPTION
**Pull request by Emmanuel0077**
…ementing abandoned first line cohort

### CODE REVIEW

**Review by TinoMuianga — COMMENTED** [review]
Updated Query of abandonned Or restarted already exists on MQ implementation, just need to map dates according to specification.

**Review by TinoMuianga — COMMENTED** [review]
Should check obs_datetime for Ficha Resumo (53)

**Review by TinoMuianga — APPROVED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java [hunk 1]
```diff
@@ -1691,6 +1690,98 @@ public CohortDefinition getMI15B2(Integer months) {
...
+   * última Consulta”).
+   * <li>Nota: “Data Última Consulta” é a data da última consulta clínica ocorrida durante o período
+   *     de revisão.
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getpatientsWhoAbandonedOrRestartedTarv() {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("All patients who abandoned or restarted tarv during the period ");
+    cd.addParameter(new Parameter("revisionEndDate", "revisionEndDate", Date.class));
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java]: Where is this parameter used?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java [hunk 2]
```diff
@@ -1691,6 +1690,102 @@ public CohortDefinition getMI15B2(Integer months) {
...
+            + "                             Max(e.encounter_datetime) AS last_encounter "
+            + "                    FROM     encounter e "
+            + "                    WHERE    e.encounter_datetime BETWEEN :startDate AND :endDate "
+            + "                    AND      e.voided=0 "
+            + "                    AND      e.encounter_type = ${6} "
+            + "                    GROUP BY e.patient_id )last_consultation "
+            + "ON         last_consultation.patient_id = p.patient_id "
+            + "WHERE      o.concept_id = ${6273} "
+            + "AND        o.value_coded IN (${1707}, "
+            + "                             ${1705}) "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java]: Put this on the same line

**Review by eliazmuchave — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java (hunk 1 above)
Comment (Emmanuel0077) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java]: In the operation startDate = revisionEnddate - 2m + 1d. If I remove it, there will be a syntax error : Cannot find parameter 'revisionEndDate' in org.openmrs.module.reporting.cohort.definition.SqlCohortDefinition All patients who abandoned or restarted tarv during the period

**Review by eliazmuchave — APPROVED** [review]
Looks good to me

**Pull request merged**